src/main/java/fi/aalto/cs/apluscourses/model/task/Task.java

```diff
@@ -0,0 +1,51 @@
+package fi.aalto.cs.apluscourses.model.task;
+
+import com.intellij.openapi.project.Project;
+import fi.aalto.cs.apluscourses.intellij.utils.ActivitiesListener;
+import fi.aalto.cs.apluscourses.utils.Event;
+import org.jetbrains.annotations.NotNull;
+
+public abstract class Task {
+
+  private String action;
+
+  @NotNull
+  public final Event taskUpdated;
+
+  /**
+   * Task constructor. Action and file are Strings
+   * read from the configuration file.
+   * @param action the action to be performed
+   */
+  protected Task(String action) {
+    this.action = action;
+    this.taskUpdated = new Event();
+  }
+
+  /**
+   * Empty Task constructor.
+   */
+  protected Task() {
+    action = "editorOpen";
+    this.taskUpdated = new Event();
+  }
+
+  public String getAction() {
+    return action;
+  }
+
+  /**
+   * Sets the current Task as complete and also triggers an Event
+   * that calls the method currentTaskCompleted in TutorialViewModel.
+   */
+  public void setIsComplete() {
+    taskUpdated.trigger();
+  }
```

Comment:
Perhaps `setComplete` would be a better name, at least to my ears "set" and "is" in the same method name sound strange/contradictory